File: frontend/src/services/adminOverview.ts
import api from "@/services/api";

export interface AdminInboxCounts {
  pending_invites: number;
  pending_companies: number;
  pending_jobs: number;
  new_applications: number;
}

export interface TopJobEntry {
  id: number;
  title: string;
  application_count: number;
}

export interface AdminStatsCounts {
  active_companies: number;
  published_jobs: number;
  total_candidates: number;
  application_status_counts: Record<string, number>;
  top_jobs: TopJobEntry[];
}

export interface AdminOverviewRead {
  inbox: AdminInboxCounts;
  stats: AdminStatsCounts;
}

export async function getAdminOverview(signal?: AbortSignal): Promise<AdminOverviewRead> {
  const res = await api.get<AdminOverviewRead>("/api/admin/overview", { signal });
  return res.data;
}

File: src/schemas/admin_overview.py
"""Admin overview response schema."""

from pydantic import BaseModel


class TopJobEntry(BaseModel):
    id: int
    title: str
    application_count: int


class AdminInboxCounts(BaseModel):
    pending_invites: int
    pending_companies: int
    pending_jobs: int
    new_applications: int


class AdminStatsCounts(BaseModel):
    active_companies: int
    published_jobs: int
    total_candidates: int
    application_status_counts: dict[str, int]
    top_jobs: list[TopJobEntry]


class AdminOverviewRead(BaseModel):
    inbox: AdminInboxCounts
    stats: AdminStatsCounts
